tools/autograd/gen_variable_type.py

```diff
@@ -945,6 +946,8 @@ def record_function_input_names():
     body.append(post_record_trace)
     if requires_derivative:
         body.append(emit_save_outputs())
+    if reset_grad_accumulator:
+        body.append('reset_grad_accumulator(self_);')
```

Comment:
nit: is it always `self_` here?
To avoid issue when other people change this, you can assert here that there is a single output and it is what you expect.